pkg/cmd/attestation/verification/sigstore.go

```diff
@@ -89,19 +66,30 @@ func (v *LiveSigstoreVerifier) chooseVerifier(b *bundle.ProtobufBundle) (*verify
 	issuer := leafCert.Issuer.Organization[0]
 
 	// if user provided a custom trusted root file path, use the custom verifier
-	if v.customVerifier != nil {
-		return v.customVerifier, issuer, nil
+	if v.config.CustomTrustedRoot != "" {
+		customVerifier, err := newCustomVerifier(v.config.CustomTrustedRoot)
```

Comment:
**Question**

I notice that the custom verifier and the github verifier branches and their `new()` functions are untested and there's actually quite a lot going on inside them.

1. How have you been manually verifying this code works?
2. Is there some way we could automate some of it?
3. If not, how would you feel about a seam for constructing verification config that can be tested. This could then be passed to the `NewSignedEntityVerifier`. I think it would reduce the untested code significantly. I can spike something to demonstrate what I mean.

Reply:
Unit testing would be a bear, but @malancas has some good integration testing in `pkg/cmd/attestation/verification/sigstore_test.go`. That was only covering `newPublicGoodVerifier()` though, so I added two tests to cover `newGitHubVerifier()` and `newCustomVerifier()`.